scipy/stats/stats.py

```diff
@@ -5718,3 +5720,175 @@ def rankdata(a, method='average'):
 
     # average method
     return .5 * (count[dense] + count[dense - 1] + 1)
+
+
+def ratio_unif(f, size=1, x1=None, x2=None, y2=None, shift=0, maxiter=None,
+               bounds=None):
+    """
+    Generate random variables using ratio-of-uniforms method.
+
+    Given a continuous univariate density f and a constant c, define the set
+    A(c) := {(x, y) : 0 < y <= sqrt(f(x/y + c)) }. If (U, V) is a random
+    vector uniformly distributed over A(c), then U/V + c follows a distribution
+    with density f.
+
+    The above result (see [1]_, [2]_) can be used to sample random variables
+    using only the density. Typical choices of c are zero or the mode of f.
+    The set A(c) is a subset of the rectangle R = [-x1, x2] x [0, y2] where
+    - x1 = inf (x - c) sqrt(f(x))
+    - x2 = sup (x - c) sqrt(f(x))
+    - y2 = sup sqrt(f(x))
+
+    In particular, these values are finite if f is bounded and x**2 * f(x) is
+    bounded (i.e. subquadratic tails). One can generate (U, V) uniformly on
+    R and return U/V + c if (U, V) are also in A(c) which can be directly
+    verified. Intuitively, this works well if A(c) fills up most of the
+    enclosing rectangle such that the probability is high that (U, V) lies in
+    A(c) whenever it lies in R(c) as the number of required iterations
+    becomes too large otherwise. To be more precise, note that the expected
+    number of iterations to draw (U, V) uniformly distributed on R(c) such that
+    (U, V) is also in A(c) is given by the ratio
+    area(R(c)) / area(A(c)) = 2 * y2 * (x2 - x1), using the fact that
+    the area of A(c) is equal to 1/2 (Theorem 7.1 in [1]_).
```

Comment:
I would relegate 5730 to 5752 to the notes. Overly mathematical for the initial part of the docstring.

Reply:
I moved some more material to the notes, however, some parts need to stay to understand the method in my view

Reply:
The text at the start of the function does need to be kept, but in this notes section.